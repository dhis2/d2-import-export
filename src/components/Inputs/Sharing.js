import i18n from '@dhis2/d2-i18n'
import React from 'react'
import { RadioGroup } from '../FinalFormComponents/RadioGroup'

export const OPTION_WITH_SHARING = {
    value: 'false',
    label: i18n.t('With sharing'),
}
export const OPTION_WITHOUT_SHARING = {
    value: 'true',
    label: i18n.t('Without sharing'),
}

export const SHARING_KEY = 'skipSharing'
export const SHARING_DEFAULT_VALUE = OPTION_WITH_SHARING.value

export const Sharing = () => (
    <RadioGroup
        name={SHARING_KEY}
        label={i18n.t('Sharing')}
        options={[OPTION_WITH_SHARING, OPTION_WITHOUT_SHARING]}
        dataTest="input-sharing"
    />
)
