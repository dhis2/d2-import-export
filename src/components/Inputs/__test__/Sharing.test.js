import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { Sharing } from '../Sharing'

jest.mock('react-final-form')

describe('Input component - Sharing', () => {
    it('should render correctly', () => {
        const file = shallow(<Sharing />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
