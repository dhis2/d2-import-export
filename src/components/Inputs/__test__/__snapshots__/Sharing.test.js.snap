// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - Sharing should render correctly 1`] = `
<Field>
  <Label>
    Sharing
  </Label>
  <RadioGroup
    dataTest="input-sharing"
    label="Sharing"
    name="skipSharing"
    options={
      Array [
        Object {
          "label": "With sharing",
          "value": "false",
        },
        Object {
          "label": "Without sharing",
          "value": "true",
        },
      ]
    }
  />
</Field>
`;
